examples/signal/plot_covariance_estimation.py

```diff
@@ -0,0 +1,205 @@
+"""
+===============================================================================
+Estimate covariance with different time windows
+===============================================================================
+
+Covariance estimators comparison for different EEG signal lengths and their
+impact on classification [1]_.
+"""
+# Author: Sylvain Chevallier
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib import pyplot as plt
+from mne import Epochs, pick_types, events_from_annotations
+from mne.io import concatenate_raws
+from mne.io.edf import read_raw_edf
+from mne.datasets import eegbci
+from pyriemann.estimation import Covariances
+from pyriemann.utils.distance import distance
+from pyriemann.classification import MDM
+from sklearn.model_selection import cross_val_score, StratifiedKFold
+from sklearn.pipeline import make_pipeline
+
+###############################################################################
+# Estimating covariance on synthetic data
+# ----------------------------------------
+#
+# Generate synthetic data, sampled from a distribution considered as the
+# groundtruth.
+
+n_trials, n_channels, n_times = 10, 5, 1000
+var = 2.0 + 0.1 * np.random.randn(n_trials, n_channels)
```

Comment:
You should fix the random seed at the beginning of the script, to have reproductible results.